src/main/drivers/pwm_output.c

```diff
@@ -193,44 +201,44 @@ void pwmCompleteOneshotMotorUpdate(uint8_t motorCount)
 
 void pwmBrushedMotorConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, uint16_t motorPwmRate)
 {
-    uint32_t hz = PWM_BRUSHED_TIMER_MHZ * 1000000;
+    const uint32_t hz = PWM_BRUSHED_TIMER_MHZ * 1000000;
     motors[motorIndex] = pwmOutConfig(timerHardware, PWM_BRUSHED_TIMER_MHZ, hz / motorPwmRate, 0);
     motors[motorIndex]->pwmWritePtr = pwmWriteBrushed;
 }
 
 void pwmBrushlessMotorConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, uint16_t motorPwmRate, uint16_t idlePulse)
 {
-    uint32_t hz = PWM_TIMER_MHZ * 1000000;
+    const uint32_t hz = PWM_TIMER_MHZ * 1000000;
     motors[motorIndex] = pwmOutConfig(timerHardware, PWM_TIMER_MHZ, hz / motorPwmRate, idlePulse);
     motors[motorIndex]->pwmWritePtr = pwmWriteStandard;
 }
 
-void pwmFastPwmMotorConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, uint8_t fastPwmProtocolType, uint16_t motorPwmRate, uint16_t idlePulse)
+void pwmFastPwmMotorConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, uint16_t motorPwmRate, uint16_t idlePulse, uint8_t fastPwmProtocolType)
 {
     uint32_t timerMhzCounter;
 
     switch (fastPwmProtocolType) {
-        default:
-        case (PWM_TYPE_ONESHOT125):
-            timerMhzCounter = ONESHOT125_TIMER_MHZ;
-            break;
-        case (PWM_TYPE_ONESHOT42):
-            timerMhzCounter = ONESHOT42_TIMER_MHZ;
-            break;
-        case (PWM_TYPE_MULTISHOT):
-            timerMhzCounter = MULTISHOT_TIMER_MHZ;
+    default:
+    case (PWM_TYPE_ONESHOT125):
+        timerMhzCounter = ONESHOT125_TIMER_MHZ;
+        motors[motorIndex]->pwmWritePtr = pwmWriteOneShot125;
+        break;
+    case (PWM_TYPE_ONESHOT42):
+        timerMhzCounter = ONESHOT42_TIMER_MHZ;
+        motors[motorIndex]->pwmWritePtr = pwmWriteOneShot42;
+        break;
+    case (PWM_TYPE_MULTISHOT):
+        timerMhzCounter = MULTISHOT_TIMER_MHZ;
+        motors[motorIndex]->pwmWritePtr = pwmWriteMultiShot;
```

Comment:
This will hard fault as motors[motorIndex] is not yet initialised (see below in the for loop - it is set by pwmOutConfig).

Need a local variable, and then set it in the for loop.